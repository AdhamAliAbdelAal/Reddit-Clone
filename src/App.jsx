import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { CssBaseline, ThemeProvider } from '@mui/material';

import MainNavBar from './components/MainNavBar/MainNavBar';
import SettingsProfile from './components/Settings/SettingsProfile/SettingsProfile';
import SettingsPrivacy from './components/Settings/SettingsPrivacy/SettingsPrivacy';
import SettingsFeed from './components/Settings/SettingsFeed/SettingsFeed';
import SettingsAccount from './components/Settings/SettingsAccount/SettingsAccount';
import Settings from './pages/Settings';

import SignUpPage from './pages/SignUpPage';
import LogInPage from './pages/LogInPage';
import ForgetUsernamePage from './pages/ForgetUsernamePage';
import ForgetPasswordPage from './pages/ForgetPasswordPage';
import ResetPasswordPage from './pages/ResetPasswordPage';

import HomePage from './pages/HomePage';
import Messages from './pages/Messages';
import Moderation from './pages/Moderation';
import Profile from './pages/Profile';
import Search from './pages/Search';
import theme from './styles/theme';
import CreatePost from './pages/CreatePost';
import PostTypeContextProvider from './contexts/PostTypeContext';
import Notifications from './pages/Notifications';
import SubReddit from './pages/SubReddit';

function App() {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <PostTypeContextProvider>
        <Router className="App">
          <MainNavBar />
          <Routes>
            <Route
              path="/"
              element={
                <HomePage data-testid="home page" />
          }
            />
            <Route
              path="/:postClass"
              element={
                <HomePage />
          }
            />
            <Route
              path="/user"
              element={
                <Profile />
              }
            />
            <Route
              path="/register"
              element={
                <SignUpPage />
}
            />
            <Route
              path="/user/:username/"
              element={
                <Profile />
          }
            >
              <Route
                path=":subTitle"
                element={
                  <Profile />
          }
              >
                <Route
                  path=":sort"
                  element={
                    <Profile />
          }
                />

              </Route>
            </Route>

            <Route
              path="/login"
              element={
                <LogInPage />
          }
            />
            <Route
              path="/password"
              element={
                <ForgetPasswordPage />
          }
            />
            <Route
              path="/resetpassword/:token"
              element={
                <ResetPasswordPage />
          }
            />

            <Route
              path="/username"
              element={
                <ForgetUsernamePage />
          }
            />
            <Route
              path="/search"
              element={
                <Search />
          }
            />
            <Route
              path="/messages"
              element={
                <Messages />
          }
            />
            <Route
              path="/moderation"
              element={
                <Moderation />
          }
            />
            <Route
              path="/submit"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/submit/:subredditName"
              element={
                <CreatePost />
            }
            />
            <Route
              path="/notifications"
              element={
                <Notifications />
          }
            />
            <Route
              path="/SubReddit/:Name"
              element={
                <SubReddit />
          }
            >
              <Route
                path=":postClass"
                element={
                  <SubReddit />
          }
              />
            </Route>
            <Route
              path="/settings"
              element={
                <Settings />
          }
            >
              <Route
                path=""
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="account"
                element={
                  <SettingsAccount />
          }
              />
              <Route
                path="profile"
                element={
                  <SettingsProfile />
          }
              />
              <Route
                path="privacy"
                element={
                  <SettingsPrivacy />
          }
              />
              <Route
                path="feed"
                element={
                  <SettingsFeed />
          }
              />
            </Route>
          </Routes>
        </Router>
      </PostTypeContextProvider>
    </ThemeProvider>
  );
}

export default App;
