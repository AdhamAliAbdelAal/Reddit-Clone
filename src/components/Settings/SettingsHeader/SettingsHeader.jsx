import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import {
  HeaderContiner, TabsContiner, Tab, TabText, Nav,
} from './styles';
/**
 * - SettingsHeader
 * - Head of Settings
 *  @component
 */
function SettingsHeader() {
  const [navigate, setNavigate] = useState('/settings');
  const history = useNavigate();
  useEffect(() => {
    setNavigate(window.location.pathname);
  }, [history]);
  return (
    <section>

      <HeaderContiner>User settings</HeaderContiner>

      <Nav>
        <TabsContiner>
          <Tab
            active={navigate === '/settings' || navigate === '/settings/account'}
            to="/settings/account"
          >
            <TabText>Account</TabText>
          </Tab>
          <Tab to="/settings/profile" active={navigate === '/settings/profile'}>
            <TabText>
              Profile
            </TabText>
          </Tab>
          <Tab to="/settings/privacy" active={window.location.pathname === '/settings/privacy'}>
            <TabText>Safety & privacy</TabText>
          </Tab>
          <Tab to="/settings/feed" active={window.location.pathname === '/settings/feed'}>
            <TabText> Feed Settings</TabText>
          </Tab>
        </TabsContiner>
      </Nav>
    </section>
  );
}

export default SettingsHeader;
