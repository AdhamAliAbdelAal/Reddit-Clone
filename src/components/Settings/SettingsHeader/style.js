import { styled } from '@mui/system';
import { Link } from 'react-router-dom';

export const HeaderContiner = styled('h3')(() => ({
  margin: 0,
  padding: '16px 20px 20px',
  fontWeight: '700',

}));
export const SettingsContiner = styled('section')(() => ({
  backgroundColor: '#fff',
  minHeight: 'calc( 100vh - 48px )',
  paddingLeft: '28px',
}));
export const TabsContiner = styled('div')(() => ({
  display: 'flex',
  flexWrap: 'wrap',
  margin: '0 auto',

  borderBottom: '1px solid #EDEFF1',
}));
export const Tab = styled(Link)(({ active, theme }) => ({
  cursor: 'pointer',
  padding: '15px 12px 12px',
  marginRight: '8px',
  display: 'flex',
  color: '#848485',
  justifyContent: 'center',
  alignItems: 'center',
  fontWeight: '600',
  textDecoration: 'none',
  borderBottom: '',
  ...((active) && {
    color: 'black',
    borderBottom: `3px solid ${theme.palette.primary.main}`,
  }),
  '&:hover': {
    color: 'black',

  },
}));
export const TabText = styled('div')(() => ({

}));
