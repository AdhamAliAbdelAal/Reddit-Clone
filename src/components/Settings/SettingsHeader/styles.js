import { styled } from '@mui/system';
import { Link } from 'react-router-dom';

export const HeaderContiner = styled('h3')(() => ({
  margin: '0 auto',
  padding: '16px 20px 20px',
  fontWeight: '700',
  maxWidth: '1200px',
}));
export const TabsContiner = styled('div')(() => ({
  display: 'flex',
  flexWrap: 'wrap',
  margin: '0 auto',
  padding: '0 20px',
  borderBottom: '1px solid #EDEFF1',
}));
export const Tab = styled(Link)(({ active, theme }) => ({
  cursor: 'pointer',
  padding: '15px 12px 12px',
  marginRight: '8px',
  display: 'flex',
  color: '#848485',
  justifyContent: 'center',
  alignItems: 'center',
  fontWeight: '600',
  textDecoration: 'none',
  borderBottom: '',
  ...((active) && {
    color: 'black',
    borderBottom: `3px solid ${theme.palette.primary?.main}`,
  }),
  '&:hover': {
    color: 'black',

  },
}));
export const TabText = styled('div')(() => ({

}));
export const Nav = styled('nav')(() => ({
  margin: '0 auto',
  maxWidth: '1200px',
}));
