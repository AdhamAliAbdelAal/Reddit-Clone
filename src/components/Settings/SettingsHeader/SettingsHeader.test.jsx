import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  render, screen,
} from '@testing-library/react';
import SettingsHeader from './SettingsHeader';
import App from '../../../App';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(
    <Router>
      <SettingsHeader />
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
// render settings-privacy page
it('should render settings-header page', () => {
  window.history.pushState({}, '', '/settings');
  render(<App />);
  expect(screen.getByTestId('settings-header')).toBeInTheDocument();
});
