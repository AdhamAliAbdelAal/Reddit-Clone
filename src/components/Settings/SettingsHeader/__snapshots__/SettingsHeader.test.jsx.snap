// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<section
  data-testid="settings-header"
>
  <h3
    className="css-5g9ivt"
    id="settings"
  >
    User settings
  </h3>
  <nav
    className="css-1ja20oy"
  >
    <div
      className="css-1vgr6oh"
    >
      <a
        active={true}
        className="css-1xyom3n"
        href="/settings/account"
        onClick={[Function]}
      >
        <div
          className="css-0"
        >
          Account
        </div>
      </a>
      <a
        active={false}
        className="css-y3du23"
        href="/settings/profile"
        onClick={[Function]}
      >
        <div
          className="css-0"
        >
          Profile
        </div>
      </a>
      <a
        active={false}
        className="css-y3du23"
        href="/settings/privacy"
        onClick={[Function]}
      >
        <div
          className="css-0"
        >
          Safety & privacy
        </div>
      </a>
      <a
        active={false}
        className="css-y3du23"
        href="/settings/feed"
        onClick={[Function]}
      >
        <div
          className="css-0"
        >
           Feed Settings
        </div>
      </a>
    </div>
  </nav>
</section>
`;
